compiler/front/options.nim

```diff
@@ -315,71 +135,78 @@ type
                           ## fields marked with '*' are subject to
                           ## the incremental compilation mechanisms
                           ## (+) means "part of the dependency"
-    backend*: TBackend ## set via `nim x` or `nim --backend:x`
-    target*: Target       # (+)
+
+    # active configuration handling
+    active*: ActiveConf
```

Comment:
```suggestion
    # REFACTOR - rename `ConfigRef` to something like `CompilerStateRef` because it's just global
    # data, then `active` can be renamed `conf`.
    active*: CurrentConf
      ## current compiler configuration based on initial inputs
```

We can leave this as `active`, but `current` doesn't sound better either. The problem is that `ConfigRef` isn't right, it's just global compiler state dump... as suggested above if you want to tackle the `ConfigRef` to something like `CompilerStateRef` then that works, in which case this might just be `conf` or something.

Personally, I think that's a big change, I'd rather leave a refactor comment here outlining it, hence my suggestion (line wrap might be off).